ansible/library/interface_facts.py

```diff
@@ -66,19 +74,20 @@ def get_default_interfaces(ip_path, module):
     )
     interface = dict(v4 = {}, v6 = {})
     for key in command.keys():
-        """
-        if key == 'v6' and self.facts['os_family'] == 'RedHat' \
+        '''
+        if key == 'v6' and self.facts['os_family'] == 'RedHat' \\
             and self.facts['distribution_version'].startswith('4.'):
             continue
-        """
+        '''
         if key == 'v6' and not socket.has_ipv6:
             continue
-        rc, out, err = module.run_command(command[key])
+        proc = subprocess.Popen(command[key], shell=False, stdout=subprocess.PIPE)
+        (out, err) = proc.communicate()
```

Comment:
What is the difference with calling run_command?

Reply:
This part of the code is now stored as a string variable in this file.
```
INTF_IP_GET_INFO_CMDs = """#!/usr/bin/python
...    .... ....
gather_ip_interface_info()
"""
```
The idea I had was to create a script named INTF_IP_GET_INFO_CMDs in the DUT when the ansible module is run. 

    # Create a python script file in the DUT.
    with open(INTF_IP_GET_INFO_SCRIPT, "w") as f:
        f.write(INTF_IP_GET_INFO_CMDs)
        f.close()

This way we could get the interface details from linux namespace as well. But when we do this way, the new script whish is created don't have the module  passed as input, hence I used the subprocess() call.

Reply:
@judyjoseph learnt this from Joe recently, if you add test=True in your Popen parameter list the output will become regular string instead of unicode encoded string, then you can remove '\\n' changes below. This is a python3 behavior change.

Reply:
Thanks @yxieca , I investigated a bit more, I have converted the string INTF_IP_GET_INFO_CMDs to a **raw** string by putting an 'r' in front of the string. The helps me not use the escape character.